iocore/net/SSLUtils.cc

```diff
@@ -1583,7 +1583,10 @@ SSLInitServerContext(const SSLConfigParams *params, const ssl_user_config &sslMu
     SSL_CTX_set_default_passwd_cb_userdata(ctx, &ud);
   }
 
-  if (sslMultCertSettings.cert) {
+  if (!sslMultCertSettings.cert && sslMultCertSettings.opt != SSLCertContext::OPT_TUNNEL) {
+    Warning("No ssl_cert_name specified and no tunnel action set");
+    goto fail;
+  } else if (sslMultCertSettings.cert) {
```

Comment:
I think this is "ok", it just reads a little odd. I probably would have preferred e.g.

    if (!sslMultCertSettings.cert) {
        if (sslMultCertSettings.opt != SSLCertContext::OPT_TUNNEL) {
            Warning( ...)
        }
    } else {
        SimpleTokenizer cert_tok((const ...

I know, a little nitpick, but the fear being that we repeat the conditional multiple time, having a risk of inconsistencies.